.github/workflows/build-linux.yml

```diff
@@ -0,0 +1,30 @@
+
+name: Build on Ubuntu
+
+on:
+  pull_request:
+    branches: [ "master" ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-24.04
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up JDK
+      uses: actions/setup-java@v4
+      with:
+        java-version: '21'
+        distribution: 'temurin'
+        cache: maven
+    - name: Install Maven
+      run: |
+        curl -s -S -o ./apache-maven-3.9.9-bin.zip https://dlcdn.apache.org/maven/maven-3/3.9.9/binaries/apache-maven-3.9.9-bin.zip
+        unzip -q ./apache-maven-3.9.9-bin.zip
+    - name: Build with Maven
+      # qa skips documentation - we check it on Jenkins CI
+      # We skip checkstyle too - we check it on Jenkins CI
+      run: ./apache-maven-3.9.9/bin/mvn -B -e -ntp install -Pstaging -Pqa '-Dcheckstyle.skip=true'
+      env:
```

Comment:
- I guess `env` might be dropped, as in most cases - PR from fork - secret will not be provided to runner (https://github.com/eclipse-ee4j/glassfish/pull/25336#discussion_r1927666235).